### PR TITLE
Feature: stop abacus manually according to the keyword in EXIT

### LINKED ISSUE #2708: Manually stop the calculation and output the results, such as STOPCAR of VASP.
### Background

It is usually difficult to converge when calculating large systems, but people do not want to give up this calculation result, hoping that ABACUS will end normally and produce a complete file.

### Describe the solution you'd like

such as STOPCAR of VASP

### Additional Context

https://www.vasp.at/wiki/index.php/STOPCAR

### PR DESCRIPTION
**Pull request by YuLiu98**
### Linked Issue
Fix #2708 

### Unit Tests and/or Case Tests for my changes
- A unit test is added for this feature.
- An integration test is added for this feature.

### What's changed?
- During calculations, if one provides a file named `EXIT`, in which a parameter `stop_ion` is set to `true`, abacus will end normally.

### CODE REVIEW

**Comment by kirk0830** [comment]
Hi, I suggest you try other names instead of "STOPCAR" because it looks like a file of VASP.
I provide here some other exmaples:

|software|file|
|---|---|
|Quantum ESPRESSO|${out_dir}/${prefix}.EXIT|
|CP2K|EXIT|
|ORCA|EXIT|

how do you think?

**Comment by YuLiu98** [comment]
> Hi, I suggest you try other names instead of "STOPCAR" because it looks like a file of VASP. I provide here some other exmaples:
> 
> software	file
> Quantum ESPRESSO	${out_dir}/${prefix}.EXIT
> CP2K	EXIT
> ORCA	EXIT
> how do you think?

How about `STOP`, `TERMINATE`, `QUIT` or some else?

**Comment by kirk0830** [comment]
> > Hi, I suggest you try other names instead of "STOPCAR" because it looks like a file of VASP. I provide here some other exmaples:
> > software	file
> > Quantum ESPRESSO	${out_dir}/${prefix}.EXIT
> > CP2K	EXIT
> > ORCA	EXIT
> > how do you think?
> 
> How about `STOP`, `TERMINATE`, `QUIT` or some else?

I guess simply "EXIT" may be okay...

File: docs/advanced/opt.md [hunk 1]
```diff
@@ -78,3 +78,12 @@ Sometimes we want to do variable-cell relaxation with some of the cell degrees o
 - ISIF = 5 : calculation = "cell-relax", fixed_axes = "volume", fixed_atoms = True
 - ISIF = 6 : calculation = "cell-relax", fixed_atoms = True
 - ISIF = 7 : calculation = "cell-realx", fixed_axes = "shape", fixed_atoms = True
+
+### Stop Geometry Optimization Manually
+
+It is usually difficult to converge when calculating large systems, but people do not want to give up this calculation result.
+Providinf a file named `EXIT`:
```
Comment (mohanchen) [inline docs/advanced/opt.md]: “Providinf" seems a typo

File: source/module_io/CMakeLists.txt [hunk 2]
```diff
@@ -14,6 +14,7 @@ list(APPEND objects
     read_cube.cpp
     read_rho.cpp
     read_rhog.cpp
+    read_stop_file.cpp
```
Comment (mohanchen) [inline source/module_io/CMakeLists.txt]: change this to read_exit_file.cpp ?

**Review by Qianruipku — APPROVED** [review]

**Pull request merged**